src/loader/README.md

```diff
@@ -2,20 +2,77 @@
 
 The loader has two main purposes:
 
-1) Load the correct versioned library for each database.
-Multiple databases in the same Postgres instance may contain
-different versions of TimescaleDB installed. The loader is
-responsible for loading the shared library corresponding
-to the correct TimescaleDB version for the database as soon
-as possible. For example, a database containing TimescaleDB
-version 0.8.0 will have timescaledb-0.8.0.so loaded.
+1) Load the correct versioned library for each database. Multiple databases in
+   the same Postgres instance may have different versions of TimescaleDB
+   installed. The loader is responsible for loading the shared library
+   corresponding to the correct TimescaleDB version for the database as soon as
+   possible. For example, a database containing TimescaleDB version 0.8.0 will
+   have timescaledb-0.8.0.so loaded.
 
-2) Starting background worker schedulers for each database.
-   Background worker schedulers launch background worker tasks
-   for TimescaleDB. The launcher is responsible for launching
-   schedulers for any database that has TimescaleDB installed.
-   This is done by a background task called the launcher.
+2) Starting background worker schedulers for each database. Background worker
+   schedulers launch background worker tasks for TimescaleDB. The launcher is
+   responsible for launching schedulers for any database that has TimescaleDB
+   installed. This is done by a background task called the launcher.
 
+# Messages the launcher may receive
+The launcher implements a simple message queue to be notified when it should
+take certain actions, like starting or restarting a scheduler for a given
+database. 
+
+##Message types sent to the launcher:
+
+`start`: Used to start the scheduler by the user. It is meant to be an
+idempotent start, as in, if it is run multiple times, it is the same as if it
+were run once. It is used mainly to reactivate a scheduler that the user had
+stopped. It does not reset the vxid of a scheduler and the started scheduler
+will not wait on txn finish.
+
+`stop`: Used to stop the scheduler immediately. It does not wait on a vxid and
+it is idempotent. 
+
+`restart`: Used to either stop and restart the scheduler if it is running or
```

Comment:
We need to update the state-machine to match this.

Reply:
done